### PR TITLE
ExpandIcon Custom Colors

### PR DESCRIPTION
**Pull request by shihaohong**
## Description

- Adds custom icon color, expanded icon color, and disabled icon colors to the ExpandIcon widget. As part of this change, ExpandIcon now uses Colors.white60 instead of Colors.white54 when dark theme is used.
- ~Updates the dark mode [ThemeData.disabledColor](https://api.flutter.dev/flutter/material/ThemeData/disabledColor.html) value to match the [Material Design specification](https://material.io/design/interaction/states.html#disabled).~ See https://github.com/flutter/flutter/issues/33149
- Fixes the Colors.white30's documented opacity value, which was incorrectly stating it was 32% opacity instead of 30%.

This takes over https://github.com/flutter/flutter/pull/31053

## Related Issues

https://github.com/flutter/flutter/issues/18992
https://github.com/flutter/flutter/pull/21328

## Tests

I added the following tests:

- Tests to verify default light/dark mode ExpandIcon's color and its disabled color
- Tests to verify ExpandIcon takes custom icon color, expanded icon color, and disabled icon color

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (Please read [Handling breaking changes]) - https://groups.google.com/forum/#!topic/flutter-announce/22w45SnuZXM
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by shihaohong** [comment]
@HansMuller I cannot tell if we're supposed to use 50% or 38% opacity for the disabled icon on dark backgrounds. There seems to have two different values:
- [MD System icon colors](https://material.io/design/iconography/system-icons.html#color) vs [MD Dark theme specs](https://material.io/design/color/dark-theme.html#ui-application)

File: packages/flutter/lib/src/material/colors.dart [hunk 1]
```diff
@@ -333,11 +332,26 @@ class Colors {
...
   ///    rather than hard-coding colors in your build methods.
-  ///  * [white, white30, white12, white10], which are variants on this color
+  ///  * [white, white50, white30, white12, white10], which are variants on this color
   ///    but with different opacities.
   static const Color white54 = Color(0x8AFFFFFF);
 
-  /// White with 32% opacity.
+  /// White with 50% opacity.
+  ///
+  /// Used for disabled icons in dark themes.
```
Comment (shihaohong) [inline packages/flutter/lib/src/material/colors.dart]: @HansMuller I cannot tell if disabledColor should be at 50% or 32% opacity for dark themes, since there are two conflicting sources of truth:
- [https://material.io/design/iconography/system-icons.html#color](MD Icon color) vs [https://material.io/design/color/dark-theme.html#ui-application](MD Dark theme)

Right now I'm thinking it might be better to stick with the dark theme recommendation?

File: packages/flutter/lib/src/material/theme_data.dart [hunk 2]
```diff
@@ -249,7 +249,7 @@ class ThemeData extends Diagnosticable {
       splashColor: splashColor,
       materialTapTargetSize: materialTapTargetSize,
     );
-    disabledColor ??= isDark ? Colors.white30 : Colors.black38;
+    disabledColor ??= isDark ? Colors.white50 : Colors.black38;
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/theme_data.dart]: This change is likely to have an impact on many apps and tests. Even though it's a tiny change, it would be good to move it to a separate PR.

**Review by HansMuller — APPROVED** [review]

File: packages/flutter/lib/src/material/colors.dart (hunk 1 above)
Comment (shihaohong) [reply packages/flutter/lib/src/material/colors.dart]: Sounds good to me, I'll make this a separate PR and revert the values for now.

Comment (shihaohong) [reply packages/flutter/lib/src/material/colors.dart]: Tracking here: https://github.com/flutter/flutter/issues/33149

File: packages/flutter/lib/src/material/theme_data.dart (hunk 2 above)
Comment (shihaohong) [reply packages/flutter/lib/src/material/theme_data.dart]: Tracking here: https://github.com/flutter/flutter/issues/33149
